### PR TITLE
Add all platforms for glslang and spirv-tools (and update)

### PR DESCRIPTION
**Pull request by SirLynix**
* Before adding new features and new modules, please go to issues to submit the relevant feature description first.
* Write good commit messages and use the same coding conventions as the rest of the project.
* Please commit code to dev branch and we will merge into master branch in feature
* Ensure your edited codes with four spaces instead of TAB.

------

* 增加新特性和新模块之前，请先到issues提交相关特性说明，经过讨论评估确认后，再进行相应的代码提交，避免做无用工作。
* 编写友好可读的提交信息，并使用与工程代码相同的代码规范，代码请用4个空格字符代替tab缩进。
* 请提交代码到dev分支，如果通过，我们会在特定时间合并到master分支上。
* 为了规范化提交日志的格式，commit消息，不要用中文，请用英文描述。

### CODE REVIEW

**Comment by SirLynix** [comment]
I tested all CI on my fork, everything works but some CI fails for connection issues randomly

File: packages/g/glslang/xmake.lua [hunk 1]
```diff
@@ -10,10 +9,15 @@ package("glslang")
...
     add_versions("1.3.231+1", "5755de46b07e4374c05fb1081f65f7ae1f8cca81")
+    add_versions("1.3.236+0", "77551c429f86c0e077f26552b7c1c0f12a9f235e")
+    add_versions("1.3.239+0", "ca8d07d0bc1c6390b83915700439fa7719de6a2a")
 
     add_configs("binaryonly", {description = "Only use binary program.", default = false, type = "boolean"})
     add_configs("exceptions", {description = "Build with exception support.", default = false, type = "boolean"})
     add_configs("rtti",       {description = "Build with RTTI support.", default = false, type = "boolean"})
+    if is_plat("wasm") then
+        add_configs("shared", {description = "Build shared library.", default = false, type = "boolean", readonly = true})
+    end
```
Comment (waruqi) [inline packages/g/glslang/xmake.lua:20]: Is it possible for wasm to use dynamic libraries? If it only supports static, I think we can disable shared for wasm inside xmake.

Comment (SirLynix) [reply packages/g/glslang/xmake.lua:20]: it does but it's a bit complicated: https://emscripten.org/docs/compiling/Dynamic-Linking.html

Comment (waruqi) [reply packages/g/glslang/xmake.lua:20]: oh, got it, thanks

**Pull request merged**